spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Gather.scala

```diff
@@ -61,16 +69,15 @@ class Gather[T: ClassTag, D: ClassTag]()(implicit ev: TensorNumeric[T], ev2: Ten
       var i = 0
       while (i < indices.nElement()) {
         val index = indices.valueAt(i + 1)
-        require(index < inputSizes(0),
-          s"index should smaller than ${inputSizes(0)}, but got $index")
-        output.select(1, i + 1).copy(inputTensor.select(1, index + 1))
+        require(index < inputSizes(dim - 1),
+          s"index should smaller than ${inputSizes(dim - 1)}, but got $index")
+        output.select(dim, i + 1).copy(inputTensor.select(dim, index + 1))
```

Comment:
line#58-line#62 and line#72-line#74 are very similar, I would suggest try to simplify and make them reusable, what do you think?